File: src/Ramanujan-Magic-Square.cpp
#include "../include/Ramanujan-Magic-Square.h"
#include "../include/Ramanujan-Magic-Square.hpp"


int main(){

int date, month, year;

// Get the dates from user
std::cout<<"Build your own magic square with your birthdate\n";
std::cout<<"Enter you Birth date(dd): ";
std::cin>>date;
std::cout<<"Enter you Birth month(mm): ";
std::cin>>month;
std::cout<<"Enter you Birth year(yyyy): ";
std::cin>>year;

std::vector<std::vector<int>> square = magic_square(date,month,year);

// Check if the returned vector is of size 1*1 (Invalid Date case)
if (square.size() == 1 && square[0].size() == 1){
std::cout<<"Invalid Date\n"; // Printing that it's a invalid date
}
else
{
std::cout<<"You Magic Square is:\n";
for(int i=0;i<4;i++){
   for(int j=0;j<4;j++){
      std::cout<<square[i][j]<<" "; // Printing the Magic square
   }
   std::cout<<std::endl;
}
}
return 0;
}

File: python/Ramanujan-Magic-Square.py
import numpy as np

Year = int(input("Enter your Birth Year(eg. 1950):"))
Month = int(input("Enter your Birth Month(eg. 01 for January):"))
Date = int(input("Enter your Birth Date(eg. 31):"))

Year_1 = int(Year/100)
Year_2 = int(Year%100)

sum = Date+Month+Year_1+Year_2

def Verification(Magic_square):
    if Magic_Square[0,0]+Magic_Square[0,1]+Magic_Square[0,2]+Magic_Square[0,3] == sum:
        if Magic_Square[1,0]+Magic_Square[1,1]+Magic_Square[1,2]+Magic_Square[1,3] == sum:
            if Magic_Square[2,0]+Magic_Square[2,1]+Magic_Square[2,2]+Magic_Square[2,3] == sum:
                if Magic_Square[3,0]+Magic_Square[3,1]+Magic_Square[3,2]+Magic_Square[3,3] == sum:
                    if Magic_Square[0,0]+Magic_Square[0,3]+Magic_Square[3,0]+Magic_Square[3,3] == sum:
                        if Magic_Square[0,0]+Magic_Square[1,1]+Magic_Square[2,2]+Magic_Square[3,3] == sum:
                            if Magic_Square[0,3]+Magic_Square[1,2]+Magic_Square[2,1]+Magic_Square[3,0] == sum:
                                if Magic_Square[0,1]+Magic_Square[0,2]+Magic_Square[3,1]+Magic_Square[3,2] == sum:
                                    if Magic_Square[1,0]+Magic_Square[2,0]+Magic_Square[1,3]+Magic_Square[2,3] == sum:
                                        if Magic_Square[0,0]+Magic_Square[0,1]+Magic_Square[1,0]+Magic_Square[1,1] == sum:
                                            if Magic_Square[0,2]+Magic_Square[0,3]+Magic_Square[1,2]+Magic_Square[1,3] == sum:
                                                if Magic_Square[1,0]+Magic_Square[1,1]+Magic_Square[2,0]+Magic_Square[2,1] == sum:
                                                    if Magic_Square[1,1]+Magic_Square[1,2]+Magic_Square[2,1]+Magic_Square[2,2] == sum:
                                                        if Magic_Square[1,2]+Magic_Square[1,3]+Magic_Square[2,2]+Magic_Square[2,3] == sum:
                                                            if Magic_Square[2,0]+Magic_Square[2,1]+Magic_Square[3,0]+Magic_Square[3,1] == sum:
                                                                if Magic_Square[2,2]+Magic_Square[2,3]+Magic_Square[3,2]+Magic_Square[3,3] == sum:
                                                                    return True

Magic_Square = np.matrix([[Date,Month,Year_1,Year_2],[Year_2+1,Year_1-1,Month-3,Date+3],[Month-2,Date+2,Year_2+2,Year_1-2],[Year_1+1,Year_2-1,Date+1,Month-1]])

if Verification(Magic_Square):
    print(Magic_Square)